src/installer/corehost/cli/hostpolicy/hostpolicy_context.cpp

```diff
@@ -166,7 +166,8 @@ int hostpolicy_context_t::initialize(hostpolicy_init_t &hostpolicy_init, const a
     }
 
     // Build properties for CoreCLR instantiation
-    pal::string_t app_base = resolver.get_app_dir();
+    pal::string_t app_base;
+    resolver.get_app_dir(&app_base);
     coreclr_properties.add(common_property::TrustedPlatformAssemblies, probe_paths.tpa.c_str());
```

Comment:
At line 129 above, we explicitly add `System.Private.CoreLib` to the TPA for non-single file or if it is not in the bundle. Do we need to keep doing that in compat mode? (I don't actually know why we need to do that in general)

Reply:
As far as I can tell the code at line 129 is not needed anymore - but it has been there since forever. So I would say we leave it as is for 5. We can definitely discuss removing it for 6 - mainly because runtime actually doesn't use TPA to look for corelib (or with my latest change it will only do so as the last resort). And as far as I can tell SDK will put corelib into `.deps.json` always anyway.

The other thing is that runtime will use the first occurrence for each assembly simple name, since we're appending here this value is basically never used for anything.

Reply:
> mainly because runtime actually doesn't use TPA to look for corelib

For 6, I would rather change the runtime to always use TPA for corelib and get rid of the other paths.

Reply:
I agree - just the bundle check and TPA - but it also needs refactoring to not parse TPA twice...